src/pam_hbac_config.c

```diff
@@ -281,13 +281,20 @@ ph_read_config(pam_handle_t *pamh,
 
     logger(pamh, LOG_DEBUG, "config file: %s", config_file);
 
+#ifdef HPUX
+    errno = 666;       // for some reason when invoked through pam in HPUX, fopen doesn't set errno on error
```

Comment:
Hmm, actually, I would also remove the errno=0 completely and added the ifdef to where we actually assign errno to ret. So on non-HP platforms, we would use:
`ret = errno`
and on HP we would use:
`ret = EIO.` We could also add a HP-specific logger() call to avoid confusing the reader with EIO always, just a generic `cannot open config file message`.